test/solc/CommandLineInterface.cpp

```diff
@@ -241,6 +241,54 @@ BOOST_AUTO_TEST_CASE(cli_input)
 	BOOST_CHECK_EQUAL(result.reader.allowedDirectories(), expectedAllowedPaths);
 }
 
+BOOST_AUTO_TEST_CASE(cli_optimizer_disabled_yul_optimization_input_blank_spaces)
+{
+	TemporaryDirectory tempDir(TEST_CASE_NAME);
+	createFilesWithParentDirs({tempDir.path() / "input.sol"});
+
+	std::string const expectedMessage =
+		"--yul-optimizations is invalid with a non-empty sequence if Yul optimizer is disabled."
+		" Note that if an empty or blank string is supplied, the default cleanup sequence is used. The empty sequence is properly denoted by \":\"."
+	;
+	std::string const blankInput = "   ";
+
+	BOOST_CHECK_EXCEPTION(
+		parseCommandLineAndReadInputFiles({
+			"solc",
+			"--strict-assembly",
```

Comment:
```suggestion
```

Reply:
It would actually not hurt to test both as long as it can be done concisely, e.g. by looping over
```c++
{
    ["--strict-assembly", "--yul-optimizations="],
    ["--strict-assembly", "--yul-optimizations=    "],
    ["--yul-optimizations="],
    ["--yul-optimizations=    "],
}
```
which would be used to construct the command. Both assembling and compilation go through the same path in command-line parser now, but it's not guaranteed that it will always be like this and the issue affects both.

I'd say it's optional, but @nikola-matic has a point that currently you're assembling with input file called `input.sol`.

Reply:
Actually, this should have been `input.yul`. I was testing using the example from the issue.
I will add the variations.